sdk/communication/azure-communication-rooms/azure/communication/rooms/_models/_models.py

```diff
@@ -3,52 +3,128 @@
 )
 
-class RoomParticipant():
+
+class ParticipantRole(str, Enum, metaclass=CaseInsensitiveEnumMeta):
+    """The role of a room participant. The default value is Attendee."""
+
+    PRESENTER = "Presenter"
+    ATTENDEE = "Attendee"
+    CONSUMER = "Consumer"
+
+
+class RoomParticipant:
     """A participant of the room.
 
     All required parameters must be populated in order to send to Azure.
 
     :ivar communication_identifier: Communication Identifier.
-    :vartype communication_identifier:~azure.communication.rooms._shared.models.CommunicationIdentifier
+    :vartype communication_identifier: ~azure.communication.rooms.CommunicationIdentifier
     :ivar role: Role Name.
     :vartype role: Union[str, ParticipantRole]
     """
-    def __init__(self,
-                *,
-                communication_identifier: CommunicationIdentifier,
-                role: Union[str, ParticipantRole] = ParticipantRole.ATTENDEE,
-                **kwargs: Any
+    @overload
+    def __init__(
+        self,
+        *,
+        communication_identifier: Union[str, CommunicationIdentifier],
+        role: Union[str, ParticipantRole] = ParticipantRole.ATTENDEE,
     ) -> None:
         """
         :keyword communication_identifier: Raw ID representation of the communication identifier. Please refer to the
          following document for additional information on Raw ID. :code:`<br>`
          https://learn.microsoft.com/azure/communication-services/concepts/identifiers?pivots=programming-language-rest#raw-id-representation.
+        :paramtype communication_identifier: str or ~azure.communication.rooms.CommunicationIdentifier
         :keyword role: The role of a room participant. The default value is Attendee. Known
          values are: "Presenter", "Attendee", and "Consumer".
         :paramtype role: str or ~azure.communication.rooms.models.ParticipantRole
         """
-        super().__init__(**kwargs)
-        self.communication_identifier = communication_identifier
-        self.role = role or ParticipantRole.ATTENDEE
-
+    @overload
+    def __init__(self, mapping: Mapping[str, Any]) -> None:
+        """
+        :param mapping: raw JSON to initialize the model. Positional only.
+        :type mapping: Mapping[str, Any]
+        """
+    
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        if len(args) == 1:
+            self.communication_identifier = identifier_from_raw_id(args[0]['rawId'])
+            self.role = args[0]['role']
+        elif args:
+            raise TypeError(f"RoomParticipant.__init__() takes 2 positional arguments but {len(args) + 1} were given.")
+        else:
+            identifier = kwargs['communication_identifier']
+            if isinstance(identifier, str):
+                self.communication_identifier = identifier_from_raw_id(identifier)
+            else:
+                self.communication_identifier = identifier
+            self.role = kwargs.get('role', ParticipantRole.ATTENDEE)
```

Comment:
this structure looks good to me: the main things we would want for forward compat with the new models is to not inherit from `msrest.Model`. This doesn't have all of the mapping support (i.e. ability to do `model["role"]`, but that can be additive down the road, so this looks good to me, thanks!